src/Functions/custom/Remove-AzFunctionApp.ps1

```diff
@@ -0,0 +1,67 @@
+function Remove-AzFunctionApp {
+    [OutputType([System.Boolean])]
+    [CmdletBinding(DefaultParameterSetName='ByName', SupportsShouldProcess=$true, ConfirmImpact='Medium')]
+    [Microsoft.Azure.PowerShell.Cmdlets.Functions.Description('Deletes a function app.')]
+    [Microsoft.Azure.PowerShell.Cmdlets.Functions.Profile('latest-2019-04-30')]
+    param(
+        [Parameter(ParameterSetName='ByName', Mandatory=$true, HelpMessage='The name of function app.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Path')]
+        [System.String]
+        [ValidateNotNullOrEmpty()]
+        ${Name},
+
+        [Parameter(ParameterSetName='ByName', Mandatory=$true)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Path')]
+        [System.String]
+        [ValidateNotNullOrEmpty()]
+        ${ResourceGroupName},
+
+        [Parameter(ParameterSetName='ByName', HelpMessage='The Azure subscription ID.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Path')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Runtime.DefaultInfo(Script='(Get-AzContext).Subscription.Id')]
+        [System.String]
+        [ValidateNotNullOrEmpty()]
+        ${SubscriptionId},
+
+        [Parameter(ParameterSetName='ByObjectInput', Mandatory=$true, ValueFromPipeline=$true)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Models.Api20180201.ISite[]]
+        [ValidateNotNull()]
+        ${InputObject},
+
+        [Parameter(HelpMessage='Returns true when the command succeeds.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        ${PassThru},
+
+        [Parameter(HelpMessage='The credentials, account, tenant, and subscription used for communication with Azure.')]
+        [Alias('AzureRMContext', 'AzureCredential')]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Azure')]
+        [System.Management.Automation.PSObject]
+        ${DefaultProfile}
+    )
+    process {
+        
+        if ($PsCmdlet.ParameterSetName -eq "ByObjectInput")
+        {            
+            if ($PSBoundParameters.ContainsKey("InputObject"))
+            {
+                $null = $PSBoundParameters.Remove("InputObject")
+            }
+
+            foreach ($input in $InputObject)
+            {
+                $functionsIdentity = CreateObjectFromPipeline -InputObject $input
```

Comment:
This is a nice workaround to allow good piping.  Note that IFunctionsIdentity should work for this without you having to manipulate the parameter set.  If not, we should file a bug for this

Reply:
I could not make it work by itself, so I added the work around. I will file a bug.